src/ln/channelmanager.rs

```diff
@@ -1188,101 +1193,104 @@ impl ChannelManager {
-										// pending, and if not that's OK, if not, the channel is on
-										// the chain and sending the HTLC-Timeout is their problem.
-										continue;
-									},
-									Ok(Some(msg)) => { fail_htlc_msgs.push(msg); },
-									Ok(None) => {
-										// Nothing to do here...we're waiting on a remote
-										// revoke_and_ack before we can update the commitment
-										// transaction. The Channel will automatically handle
-										// building the update_fail_htlc and commitment_signed
-										// messages when we can.
-										// We don't need any kind of timer here as they should fail
-										// the channel onto the chain if they can't get our
-										// update_fail_htlc in time, its not our problem.
 									}
-								}
-							},
+								},
+							}
 						}
-					}
 
-					if !add_htlc_msgs.is_empty() || !fail_htlc_msgs.is_empty() {
-						let (commitment_msg, monitor) = match forward_chan.send_commitment() {
-							Ok(res) => res,
-							Err(e) => {
-								if let ChannelError::Ignore(_) = e {
-									panic!("Stated return value requirements in send_commitment() were not met");
-								}
-								//TODO: Handle...this is bad!
+						if !add_htlc_msgs.is_empty() || !fail_htlc_msgs.is_empty() {
+							let (commitment_msg, monitor) = match chan.get_mut().send_commitment() {
+								Ok(res) => res,
+								Err(e) => {
+									if let ChannelError::Ignore(_) = e {
+										panic!("Stated return value requirements in send_commitment() were not met");
+									}
+									//TODO: Handle...this is bad!
+									continue;
+								},
+							};
+							if let Err(e) = self.monitor.add_update_monitor(monitor.get_funding_txo().unwrap(), monitor) {
+								handle_errors.push((chan.get().get_their_node_id(), handle_monitor_err!(self, e, channel_state, chan, RAACommitmentOrder::CommitmentFirst, false, true)));
 								continue;
-							},
-						};
-						if let Err(_e) = self.monitor.add_update_monitor(monitor.get_funding_txo().unwrap(), monitor) {
-							unimplemented!();
+							}
+							channel_state.pending_msg_events.push(events::MessageSendEvent::UpdateHTLCs {
+								node_id: chan.get().get_their_node_id(),
+								updates: msgs::CommitmentUpdate {
+									update_add_htlcs: add_htlc_msgs,
+									update_fulfill_htlcs: Vec::new(),
+									update_fail_htlcs: fail_htlc_msgs,
+									update_fail_malformed_htlcs: Vec::new(),
+									update_fee: None,
+									commitment_signed: commitment_msg,
+								},
+							});
 						}
-						channel_state.pending_msg_events.push(events::MessageSendEvent::UpdateHTLCs {
-							node_id: forward_chan.get_their_node_id(),
-							updates: msgs::CommitmentUpdate {
-								update_add_htlcs: add_htlc_msgs,
-								update_fulfill_htlcs: Vec::new(),
-								update_fail_htlcs: fail_htlc_msgs,
-								update_fail_malformed_htlcs: Vec::new(),
-								update_fee: None,
-								commitment_signed: commitment_msg,
-							},
-						});
+					} else {
```

Comment:
Build without unreachable branch (1.22.0)

Reply:
Ah astute! Maybe add a comment there with your answer

Reply:
I'm actually gonna leave it. We use this pattern in a few places, and hopefully "unreachable!()" is rather self-documenting. In general, across the library, we'd prefer to panic than have a potentially money-losing bug, so this isn't particularly out of the ordinary.